### PR TITLE
fix for qdma 2rp to work

### PR DESCRIPTION
**Pull request by xuhz**
1. add build_priv for msix
2. add resource for stm to qdma

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 1]
```diff
@@ -1695,15 +1696,26 @@ static int qdma_probe(struct platform_device *pdev)
...
 	platform_set_drvdata(pdev, qdma);
 
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-	if (!res) {
-		xocl_err(&pdev->dev, "Empty resource");
-		return -EINVAL;
+	for (i = 0, res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+		res;	
+		res = platform_get_resource(pdev, IORESOURCE_MEM, ++i)) {
+		if (!strncmp(res->name, NODE_QDMA, strlen(NODE_QDMA))) {
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: res->name is NULL for legacy 2RP device. Please add name to XOCL_RES_QDMA in devices.h

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 2]
```diff
@@ -1715,7 +1727,7 @@ static int qdma_probe(struct platform_device *pdev)
 	conf->qsets_max = QDMA_QSETS_MAX;
 	conf->bar_num_config = dma_bar;
 	conf->bar_num_stm = XDEV(xdev)->bar_idx;
-	conf->stm_reg_base = 0x02000000;
+	conf->stm_reg_base = stm_base;
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: please use xocl_ioaddr_to_baroff() to get stm bar index. you may reference dma_bar

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 3]
```diff
@@ -1695,15 +1696,26 @@ static int qdma_probe(struct platform_device *pdev)
...
+		} else {
+			xocl_err(&pdev->dev, "Unknown resource: %s", res->name);
+			return -EINVAL;
+		}
 	}
 
-	ret = xocl_ioaddr_to_baroff(xdev, res->start, &dma_bar, NULL);
-	if (ret) {
-		xocl_err(&pdev->dev, "Invalid resource %pR", res);
+	if (dma_bar == -1 || stm_base == -1) {
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: legacy QDMA device does not have stm defined. Please add STM to XOCL_RES_QDMA. Please verify legacy QDMA platform.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c (hunk 1 above)
Comment (xuhz) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: will change

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c (hunk 3 above)
Comment (xuhz) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: will change

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c (hunk 2 above)
Comment (xuhz) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c]: changed. verified u200 legacy qdma

**Pull request merged**